### PR TITLE
Add LanceDB Vector store

### PR DESCRIPTION
**Pull request by AyushExel**
Great work on this repo @yyiilluu and team!

Lancedb is open source, serverless, setup free vectordb. This integration will allow using lancedb as base search tool without any additional steps, or setup.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -43,6 +43,7 @@ colorama = ">=0.4.6"
 pydantic = "^1.10.9"
 google-api-python-client = {version = ">=2.89.0", optional = true}
 chromadb = ">=0.3.26"
+lancedb = ">=0.1.16"
```
Comment (yyiilluu) [inline pyproject.toml:46]: can you move this to `[tool.poetry.extras]` section? i am planning to move chromadb and pinecone to there as well to make the package smaller

**Review by yyiilluu — COMMENTED** [review]
thank you!, happy to add another one here. just minor comment

File: pyproject.toml (hunk 1 above)
Comment (AyushExel) [reply pyproject.toml]: @yyiilluu done! followed this https://python-poetry.org/docs/pyproject/#extras

**Review by yyiilluu — APPROVED** [review]

**Comment by yyiilluu** [comment]
@AyushExel can you try to resolve this 
```
Run python -m poetry install
Skipping virtualenv creation, as specified in config file.
Installing dependencies from lock file
Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.

Because autochain depends on lancedb (>=0.1.16) which doesn't match any versions, version solving failed.
Error: Process completed with exit code 1.
```

**Comment by AyushExel** [comment]
I've tried to update the lockfile with the same poetry version it was last updated with (1.4.2). Hopefully, this resolves the issue

**Comment by yyiilluu** [comment]
thanks for your contribution!

**Pull request merged**